doc/source/reference/indexing.rst

```diff
@@ -93,7 +93,6 @@ Compatibility with MultiIndex
    :toctree: api/
 
    Index.set_names
-   Index.is_lexsorted_for_tuple
```

Comment:
As mentioned a few lines above, this is apparently for compatibility with MultiIndex. That is no longer needed?

Reply:
It is not needed by pandas maybe, but it is part of the public API (to make the API of Index and MultiIndex more alike)

Reply:
I don't think its a big deal to remove this method. Could have a note about it I guess.

Reply:
> Should we then remove it from MultiIndex as well?

It is not overridden by MultiIndex

Reply:
> It is not overridden by MultiIndex

That's not my point. What I am trying to say is: in the past, we tried to keep a consistent api between Index / MultiIndex. So why not do that here?(either keep both, or remove both)

Reply:
I'm definitely confused.  If this PR is merged, then there will be no `is_lexsorted_for_tuple` on _any_ class.  Which should classify as "remove both", shouldn't it?

Reply:
Sorry, I mis-interpreted your "this is not overriden". I was only thinking about the implications for the user (the fact that it is overriden or not does not necessarily mean this method was not there for API compatibility), but of course that means you also removed the method for MultiIndex. So fine with this!